pydarn/plotting/rtp.py

```diff
@@ -0,0 +1,988 @@
+                - 'search.noise': search noise
+                - 'sky.noise': sky noise
+                - 'nave': number of averages
+                - 'tfreq': transmission frequency
+                - 'p_l': Signal to Noise
+                - 'v': velocity
+                - 'w_l': spectral width
+                - 'elv': elevation
+            with a tuple as the value (zmin, zmax) for the plots
+            Default: {'noise.search': (1e0, 1e5),
+                      'noise.sky': (1e0, 1e5),
+                      'tfreq': (8, 22),
+                      'nave': (0, 60),
+                      'p_l': (0, 30),
+                      'v': (-200, 200),
+                      'w_l': (0, 150),
+                      'elv': (0, 50)}
+        plot_elv: boolean
+            boolean determines if elevation should be plotted or not.
+            If there is no data for elevation data field then elevation is not
+            plotted.
+            Default: True
+        title: str
+            title of the plot
+            Default: auto-generated by the files details
+            {radar name} Fitacf {version} {start_date} - {end_date}  Beam {num}
+        Raises
+        ------
+        IndexError
+        RTPUnknownParameterError
+        RTPIncorrectPlotMethodError
+        RTPNoDataFoundError
+
+        See Also
+        --------
+        plot_range_time : plots range-time parameter
+        plot_time_series : plots time-series
+
+        Return
+        ------
+        fig: matplotlib.pyplot.figure object
+        axes: list
+            list of matplotlib.pyplot.axes objects to generate
+            the subplots in the summary plot
+        """
+        message = "WARNING: matplotlib Default dpi may cause distortion"\
+                  " in range gates and time period. The figure size can"\
+                  " be adjusted with the option figsize and dpi can be"\
+                  " adjusted when saving the file."
+        warnings.warn(message)
+
+        # Default boundary ranges for the various parameter
+        boundary_ranges = {'noise.search': (1e0, 1e5),
+                           'noise.sky': (1e0, 1e5),
+                           'tfreq': (8, 22),
+                           'nave': (0, 60),
+                           'p_l': (0, 30),
+                           'v': (-200, 200),
+                           'w_l': (0, 150),
+                           'elv': (0, 50)}
+        boundary_ranges.update(boundary)
+
+        # Default color maps for the summary plot
+        line = {'noise.search': 'k',
+                'noise.sky': 'k',
+                'tfreq': 'k',
+                'nave': 'k'}
+
+        if isinstance(lines, str):
+            line.update({k: lines for k,v in line.items()})
+        else:
+            line.update(lines)
+
+        cmap = {'p_l': 'viridis',
+                'v': 'viridis',
+                'w_l': 'viridis',
+                'elv': 'viridis'}
+        if isinstance(cmaps, str):
```

Comment:
So....

This will fail if you have imported `unicode_literal`.  Probably better to test for the dict and let the mapping fail if the user puts in something horrible by mistake.

Reply:
Watch it, you might be showing some of your python 2 hairs : p Because python 2 and 3 handle strings different specifically UTF encoding. `unicode_literals` was developed to help python 2/3 backward compatibility. However, if a user was to use this and try to run my code in python 2 they will have a lot more problems. 

If we look at python3, however, 
you cannot directly import it:
```python
import unicode_literals 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'unicode_literals'
```
If I use the `__future__` statement
```python 
>>> from __future__ import unicode_literals 
>>> name = 's'
>>> type(name)
<class 'str'>
>>> name = u's'
>>> type(name)
<class 'str'>
>>> name = u"s"
>>> type(name)
<class 'str'>
>>> name = "s".encode("utf-8")
>>> type(name)
<class 'bytes'>
```
So in this example the thing I would have to be concerned about would be bytes object and the user needs to deliberatly `encode` it to get that specific object. Since python isn't type set doing a million type checks is not very pythonic. So if does come up I can look at other possibilities. I say this because there is other `str` checks that cannot be changed like this one. 

For the sake of the feedback, however, I did reverse it as your suggestion, thank you for pointing out the potential issue with `bytes` :+1: